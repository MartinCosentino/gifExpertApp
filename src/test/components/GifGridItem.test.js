import React from 'react';
const { shallow } = require("enzyme");
const { GifGridItem } = require("../../components/GifGridItem");


describe('Prueba GifGridItem', () => {

    const title = 'Un title';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />)


    test('debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });




    test('debe tener un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    });

    test('Debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');

        // console.log(img.props())

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);



    });

    test('debe de tener la misma clase', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');


        //para verificar una clase usar props
        expect(div.prop('className')).toBe('card animate__animated animate__bounce');
        expect(className.includes('animate__bounce')).toBe(true);
    })


});
