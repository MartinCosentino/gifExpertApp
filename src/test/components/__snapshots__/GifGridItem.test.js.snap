// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un title"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un title
     
  </p>
</div>
`;
